### PR TITLE
Use preprocessor for DDR_VM code

### LINKED ISSUE #13024: NoClassDefFoundError: openj9.dtfj module conflicts with uses of ibmjzos.jar on platforms other than z/OS
Hi OpenJ9 team,

on z/OS, the JZOS classes are part of the **ibm.jzos** Java module, just as expected. On platforms other than z/OS, OpenJ9 only contains a stub of the `com.ibm.jzos.ZFile` class, which is part of the **openj9.dtfj** Java module. This leads to problems when JZOS is being used on platforms other than z/OS.

Please note that this is **not** a hypothetical use case - there is at least one customer that I am aware of who actually makes use of some JZOS classes on a non-z/OS operating system.

On OpenJ9 versions up to and including **0.24.0**, the following test works as expected:

```
marc@localhost:~$ jdk-11.0.10+9/bin/java -cp /path/to/ibmjzos.jar com.ibm.jzos.FileFactory
Error: Main method not found in class com.ibm.jzos.FileFactory, please define the main method as:
   public static void main(String[] args)
or a JavaFX application class must extend javafx.application.Application
```
Whereas on OpenJ9 versions **> 0.24.0**, the same test leads to an error:

```
marc@localhost:~$ jdk-11.0.11+9/bin/java -cp /path/to/ibmjzos.jar com.ibm.jzos.FileFactory
Error: Could not find or load main class com.ibm.jzos.FileFactory
Caused by: java.lang.ClassNotFoundException: com.ibm.jzos.FileFactory
```

The root cause of this problem seems to be the inclusion of the `com.ibm.jzos.ZFile` class into the **openj9.dtfj** Java module, which dictates the visibility of the `com.ibm.jzos` package.

Is there a way to restore the old behavior?

Cheers,
Marc

### PR DESCRIPTION
**Pull request by keithc-ca**
* update preprocessor to recognize '-noWarnIncludeIf' option
* add DDR_VM preprocessor configuration
* update DDR_VM code with necessary preprocessor conditions
* remove stub implementation of ZFile

Required by (and must be merged concurrently with):
* ibmruntimes/openj9-openjdk-jdk8#513
* ibmruntimes/openj9-openjdk-jdk11#415
* ibmruntimes/openj9-openjdk-jdk16#62
* ibmruntimes/openj9-openjdk-jdk17#1
* ibmruntimes/openj9-openjdk-jdk#332

Fixes #13024.

### CODE REVIEW

File: sourcetools/com.ibm.jpp.preprocessor/com/ibm/jpp/commandline/CommandlineBuilder.java [hunk 1]
```diff
@@ -209,6 +209,7 @@ private static void missingValueFor(String arg) {
 		validArgs.add("-verdict");
 		validArgs.add("-includeifunsure");
 		validArgs.add("-nowarn");
+		validArgs.add("-nowarnincludeif");
```
Comment (pshipton) [inline sourcetools/com.ibm.jpp.preprocessor/com/ibm/jpp/commandline/CommandlineBuilder.java:212]: Are there changes missing? Seems more than this is required to implement the new option.

Comment (pshipton) [reply sourcetools/com.ibm.jpp.preprocessor/com/ibm/jpp/commandline/CommandlineBuilder.java:212]: Keith pointed out the option is already implemented at https://github.com/eclipse-openj9/openj9/blob/master/sourcetools/com.ibm.jpp.preprocessor/com/ibm/jpp/om/Builder.java#L476

Comment (keithc-ca) [reply sourcetools/com.ibm.jpp.preprocessor/com/ibm/jpp/commandline/CommandlineBuilder.java:212]: The recognition of that option is here: https://github.com/eclipse-openj9/openj9/blob/master/sourcetools/com.ibm.jpp.preprocessor/com/ibm/jpp/om/Builder.java#L476.

**Comment by keithc-ca** [comment]
Fixed `Dump.java` so code is correct when preprocessor is not used (and comments are ignored).

File: debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/tdump/zebedee/dumpreader/Dump.java [hunk 2]
```diff
@@ -168,19 +168,26 @@ public Dump(String filename, boolean initialize) throws FileNotFoundException, F
             raf = new FileImageInputStream(new RandomAccessFile(filename, "r"));
             fileSize = new File(filename).length();
         } catch (FileNotFoundException e) {
-            if (System.getProperty("os.arch").indexOf("390") == -1) {
+            /*
+             * Note: preprocessor conditions do not apply to internal builds so
```
Comment (pshipton) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/tdump/zebedee/dumpreader/Dump.java:172]: I think this should just be explicit and say IBM Java 8.

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/tdump/zebedee/dumpreader/Dump.java:172]: Sure, I can update the comment.

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/tdump/zebedee/dumpreader/Dump.java:172]: Does this work for you?
```
   * Note: IBM Java 8 builds do not use the preprocessor so the [IF] condition
   * is ignored: This must have the proper control flow ignoring those comments.
```

Comment (pshipton) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/tdump/zebedee/dumpreader/Dump.java:172]: Sure. Pls put it in TDumpReader as well.

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/tdump/zebedee/dumpreader/Dump.java]: Done.

**Comment by pshipton** [comment]
jenkins compile zlinux jdk8 depends ibmruntimes/openj9-openjdk-jdk8#513

**Comment by pshipton** [comment]
jenkins compile zlinux jdk11 depends ibmruntimes/openj9-openjdk-jdk11#415

**Comment by pshipton** [comment]
jenkins compile zlinux jdk16 depends ibmruntimes/openj9-openjdk-jdk16#62

**Comment by pshipton** [comment]
jenkins compile zlinux jdk8 depends ibmruntimes/openj9-openjdk-jdk8#513 depends eclipse/omr#master

**Comment by pshipton** [comment]
jenkins compile zlinux jdk8 depends ibmruntimes/openj9-openjdk-jdk8#513 eclipse/omr#master

**Comment by pshipton** [comment]
jenkins compile zlinux jdk11 depends ibmruntimes/openj9-openjdk-jdk11#415 eclipse/omr#master

**Comment by pshipton** [comment]
jenkins compile zlinux jdk16 depends ibmruntimes/openj9-openjdk-jdk16#62 eclipse/omr#master

**Comment by keithc-ca** [comment]
The build failures suggest the [excludes in CompileJavaModules.gmk](https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/closed/custom/CompileJavaModules.gmk#L26) are being ignored.

**Comment by pshipton** [comment]
jenkins compile xlinux,plinux,win,osx jdk11 depends ibmruntimes/openj9-openjdk-jdk11#415 eclipse/omr#master

**Comment by pshipton** [comment]
I tried to build with criu enabled, but the criu library isn't on the build machine https://github.com/eclipse-openj9/openj9/issues/12817. @jdekonin did the library get installed anywhere yet?

**Comment by pshipton** [comment]
Oops, wrong PR for the previous comment. I'll delete it.

**Comment by pshipton** [comment]
jenkins compile zlinux jdk11 depends ibmruntimes/openj9-openjdk-jdk11#415 eclipse/omr#master

**Comment by pshipton** [comment]
jenkins compile zlinux jdk16 depends ibmruntimes/openj9-openjdk-jdk16#62 eclipse/omr#master

**Review by pshipton — APPROVED** [review]

**Comment by pshipton** [comment]
Ran a DDR test using grinders on each jdk8, jdk11, jdk16. These passed.
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/82
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/83
https://openj9-jenkins.osuosl.org/view/Test/job/Grinder/84

**Comment by pshipton** [comment]
Pls cherry pick for 0.27

**Comment by keithc-ca** [comment]
> Pls cherry pick for 0.27

See #13076.

**Pull request merged**